### DISCUSSION TITLE
Information regarding training set used for Building Footprints  model

### DISCUSSION THREAD

**Discussion by traviss01** [opengeos/geoai#248] [Q&A] [answered]
Trying to figure out what the images specs were for the model? The ESRI footprint model works really well with my 10cm RGB dataset, except that legally I am not allowed to use it. 

The model in the geoai dataset does not produce anything usable for me, and I understand I can train my own dataset, and I am in the process of trying that, but I am curious if there is documentation about the inputs that went into creating your existing model, number of images, resolution etc. I am fairly new to mask r-cnn in general, I have access to quite a few fairly accurate building footprints as a shapefile, along with the imagery but I am unsure how many images, etc should go into this.

Thanks

**Comment by giswqs** [comment, marked as answer]
I previously used Esri’s pretrained building-detection model in this package. Per Esri’s request, I have removed those pretrained weights from Hugging Face as they don't allow using their pretained models outside ArcGIS. Those pretained models used to be publicly available for download, but now they require a login to download. 

The model currently on Hugging Face (.pth) was trained only on a single NAIP scene using the example notebook here: https://opengeoai.org/examples/train_building_footprints_usa. As a result, it may not generalize well to other regions.

If you need stronger performance, you can:

-  Train your own model using the notebook above—results improve with more images and higher-quality labels.
-  Extract a .pth from an Esri .dlpk and use it with the package (at your own risk).

My priority is to provide simple, reproducible workflows; I don’t have the resources to train and host large and pre-trained models at this time.

Thanks for understanding,
